builtin/logical/transit/path_decrypt.go

```diff
@@ -170,6 +177,18 @@ func (b *backend) pathDecryptWrite(ctx context.Context, req *logical.Request, d
 	}
 
 	p.Unlock()
+
+	// Depending on the errors in the batch, different status codes should be returned. User errors
```

Comment:
Shouldn't non-batch requests should benefit from this logic? The return statement in line 172 unconditionally results in a Bad Request response code, even though the error might just have been an internal one?

Reply:
Nice catch, I completely overlooked that case.

Reply:
Both encrypt and decrypt should now return an internal error if one was encountered in outside of batch input.